config/admin-functions/lockdown.sh

```diff
@@ -17,25 +25,45 @@ echo "$(cat /vx/admin/config/cmdline)${HASH}" > /tmp/cmdline
 
 KERNEL_VERSION=`uname -r`
 
-# Now package up our kernel, cmdline, etc
-objcopy \
-    --add-section .osrel="/usr/lib/os-release" --change-section-vma .osrel=0x20000 \
-    --add-section .cmdline="/tmp/cmdline" --change-section-vma .cmdline=0x30000 \
-    --add-section .splash="/vx/admin/config/logo.bmp" --change-section-vma .splash=0x40000 \
-    --add-section .linux="/boot/vmlinuz-${KERNEL_VERSION}" --change-section-vma .linux=0x2000000 \
-    --add-section .initrd="/boot/initrd.img-${KERNEL_VERSION}" --change-section-vma .initrd=0x3000000 \
-    "/usr/lib/systemd/boot/efi/linuxx64.efi.stub" "/tmp/linux.efi"
-
-# Sign the resulting binary
-# First make sure the drive is mounted
-umount /dev/sda || true
-umount /dev/sda1 || true
-mount /dev/sda /mnt || mount /dev/sda1 /mnt || (echo "Secure boot keys not found; exiting" && sleep 5 && exit);
-
-sbsign --key=/mnt/DB.key --cert=/mnt/DB.crt --output /boot/efi/EFI/debian/VxLinux-signed.efi /tmp/linux.efi
-
-# Now install it 
-bash "${VX_FUNCTIONS_ROOT}/setup-boot-entry.sh"
+
+# Now package up our kernel, cmdline, etc, if we're not on a Surface
+if [ $surface == 0 ]; then
+    objcopy \
+        --add-section .osrel="/usr/lib/os-release" --change-section-vma .osrel=0x20000 \
+        --add-section .cmdline="/tmp/cmdline" --change-section-vma .cmdline=0x30000 \
+        --add-section .splash="/vx/admin/config/logo.bmp" --change-section-vma .splash=0x40000 \
+        --add-section .linux="/boot/vmlinuz-${KERNEL_VERSION}" --change-section-vma .linux=0x2000000 \
+        --add-section .initrd="/boot/initrd.img-${KERNEL_VERSION}" --change-section-vma .initrd=0x3000000 \
+        "/usr/lib/systemd/boot/efi/linuxx64.efi.stub" "/tmp/linux.efi"
+
+    # Sign the resulting binary
+    # First make sure the drive is mounted
+    umount /dev/sda || true
+    umount /dev/sda1 || true
+    mount /dev/sda /mnt || mount /dev/sda1 /mnt || (echo "Secure boot keys not found; exiting" && sleep 5 && exit);
+
+    sbsign --key=/mnt/DB.key --cert=/mnt/DB.crt --output /boot/efi/EFI/debian/VxLinux-signed.efi /tmp/linux.efi
+
+    # Now install it 
+    bash "${VX_FUNCTIONS_ROOT}/setup-boot-entry.sh"
+else
+    # On a surface we just need to setup the right GRUB entry
+    chmod +w /boot/grub/grub.cfg
+    cp /vx/admin/config/grub.cfg /boot/grub/grub.cfg
```

Comment:
should we make it not writable after this, or maybe just not `chmod` and do a `sudo cp` which ... IIRC preserves the target permissions? I could be wrong on that.

Reply:
this whole script is run with sudo. We'd have to chmod -w after the write, which we can do.